consensus/istanbul/backend/handler.go

```diff
@@ -290,3 +298,191 @@ func (sb *Backend) UnregisterPeer(peer consensus.Peer, isProxiedPeer bool) {
 		}
 	}
 }
+
+// Handshake allows the initiating peer to identify itself as a validator
+func (sb *Backend) Handshake(peer consensus.Peer, peerIsInbound bool) (bool, error) {
```

Comment:
I'm wondering if this handshake can be done only if it's an attempted validator connection.

Reply:
If I'm a validator and the new peer initiated the connection & wants to prove itself as a validator, I have no idea if this new peer is a full node or a validator, so I'd have to listen for a message from them to decide what kind of node they are. If the peer never sent a message the handshake would timeout (which is right now considered an error)

I don't think there's a straightforward way around this, let me know if you think otherwise

Reply:
Can non validators send a "no-op" message?

Reply:
That's what I consider the empty `istanbul.Message` that's sent to be